platform/ios/test/MBXLoggingConfigurationTests.m

```diff
@@ -0,0 +1,165 @@
+#import <XCTest/XCTest.h>
+@import Mapbox;
+#import "MGLLoggingConfiguration_Private.h"
+
+@interface MBXLoggingConfigurationTests : XCTestCase
+@property (nonatomic, assign) MGLLoggingLevel loggingLevel;
+@end
+
+@implementation MBXLoggingConfigurationTests
+
+- (void)setUp {
+    self.loggingLevel = [MGLLoggingConfiguration sharedConfiguration].loggingLevel;
+
+    [super setUp];
+}
+
+- (void)tearDown {
+    [super tearDown];
+
+    [MGLLoggingConfiguration sharedConfiguration].handler = nil;
+    [MGLLoggingConfiguration sharedConfiguration].loggingLevel = self.loggingLevel;
+}
+
+- (void)internalTestLoggingLevel:(MGLLoggingLevel)level
+           expectingLogForLevels:(NSArray*)expectedLevels {
+
+    NSUInteger expectedCount = expectedLevels.count;
+
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      [NSString stringWithFormat:@"Logging expecting %lu levels", expectedCount]];
+    if (expectedCount > 0) {
+        expectation.expectedFulfillmentCount = expectedCount;
+    }
+    else {
+        // If there are no expected levels, we don't expect any logs
+        expectation.inverted = YES;
+    }
+
+    [MGLLoggingConfiguration sharedConfiguration].loggingLevel = level;
+
+    [MGLLoggingConfiguration sharedConfiguration].handler = ^(MGLLoggingLevel loggingLevel, NSString * _Nonnull filePath, NSUInteger line, NSString * _Nonnull message) {
+        NSLog(@"(%lu) Level %ld: %@", (unsigned long)line, (long)loggingLevel, message);
+        XCTAssertEqualObjects(message, @(loggingLevel).stringValue);
+        XCTAssert([expectedLevels containsObject:@(loggingLevel)]);
+        [expectation fulfill];
+    };
+
+    MGLLogInfo(@"%d", MGLLoggingLevelInfo);
+    MGLLogWarning(@"%d", MGLLoggingLevelWarning);
+    MGLLogError(@"%d", MGLLoggingLevelError);
+    MGLLogFault(@"%d", MGLLoggingLevelFault);
+    MGLLogDebug(@"%d", MGLLoggingLevelDebug);
+
+    [self waitForExpectations:@[expectation] timeout:0.1];
+}
+
+- (void)testLoggingLevelNone {
+    [self internalTestLoggingLevel:MGLLoggingLevelNone expectingLogForLevels:@[]];
+}
+
+- (void)testLoggingLevelDebug {
+    [self internalTestLoggingLevel:MGLLoggingLevelDebug
+             expectingLogForLevels:@[
+        @(MGLLoggingLevelDebug),
+        @(MGLLoggingLevelInfo),
+        @(MGLLoggingLevelWarning),
+        @(MGLLoggingLevelError),
+        @(MGLLoggingLevelFault)
+    ]];
+}
+
+- (void)testLoggingLevelInfo {
+    [self internalTestLoggingLevel:MGLLoggingLevelInfo
+             expectingLogForLevels:@[
+//      @(MGLLoggingLevelDebug),
```

Comment:
nit: please remove commented code.

Reply:
Tempted to leave for the moment, as in this case it's kinda documentation when seen alongside the other tests.